### DISCUSSION TITLE
Check if AC should be disconnected? (Poorman's AlDente)

### DISCUSSION THREAD

**Discussion by dickiedyce** [johnlindquist/kit#731] [Share]
I have a MacBook Retina, 12-inch, 2017, which makes using a tool like AlDente (for keeping the battery in good condition) impossible.

This is a scheduled script that notifies me to manually unplug when I reach 80% charge, and again manually plug it back in when the charge drops to 40%.

[Open monitor-battery in Script Kit](https://scriptkit.com/api/new?name=monitor-battery&url=https://gist.githubusercontent.com/johnlindquist/3c21deb6d0d9dd9d06d8268fdd4baa30/raw/5837f8365aab810906323fbd0430e32f544ee7db/monitor-battery.js")

```js
// Menu: Battery Monitor
// Description: Check if AC should be disconnected? Poorman's AlDente
// Author: Richard Dyce
// Twitter: @dickiedyce
// schedule: 0,15,20,45 5-22 * * *

import "@johnlindquist/kit"

const result = await exec(`pmset -g batt`)

const charge = 1 * result.stdout.match(/(\d+)\%/)[1]
const state = result.stdout.match(/(discharging|charging)/)[0]
const source = result.stdout.match(/(Battery Power|AC Power)/)[0]

if(charge > 80 && source == 'AC Power' ) {
    notify(`UNPLUG! The laptop is ${state} from ${source}, at ${charge}%`)
} else if (charge < 40 && source == 'Battery Power') {
    notify(`PLUG-IN! The laptop is ${state} from ${source}, at ${charge}%`)
} else {
   // notify(`The laptop is ${state} from ${source}, at ${charge}%`)
}
```

_Future plans: I have a Wemo WiFI Smart Plug on order, so if I can't stop the battery charging, I should be able to turn the charger off at the wall. ;-)_

**Comment by johnlindquist** [comment]
Very clever! I love seeing people think outside the box of saving files and posting to APIs. Very nicely done 👏 

And thanks for sharing!
